src/main/java/com/yugabyte/sample/apps/MultiTableMultiIndexInserts.java

```diff
@@ -0,0 +1,382 @@
+// Copyright (c) YugaByte, Inc.  //
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.  You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied.  See the License for the specific language governing permissions and limitations
+// under the License.
+//
+package com.yugabyte.sample.apps;
+
+import java.sql.Array;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.Statement;
+import java.util.*;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
+
+import org.apache.log4j.Logger;
+
+/**
+ * This workload writes and reads some random string keys from a postgresql table.
+ */
+public class MultiTableMultiIndexInserts extends AppBase {
+  private static final Logger LOG = Logger.getLogger(MultiTableMultiIndexInserts.class);
+
+  // Static initialization of this workload's config. These are good defaults for getting a decent
+  // read dominated workload on a reasonably powered machine. Exact IOPS will of course vary
+  // depending on the machine and what resources it has to spare.
+  static {
+    // Disable the read-write percentage.
+    appConfig.readIOPSPercentage = -1;
+    // Set the read and write threads to 2 each.
+    appConfig.numReaderThreads = 2;
+    appConfig.numWriterThreads = 2;
+    // The number of keys to read.
+    appConfig.numKeysToRead = -1;
+    // The number of keys to write. This is the combined total number of inserts and updates.
+    appConfig.numKeysToWrite = -1;
+    // The number of unique keys to write. This determines the number of inserts (as opposed to
+    // updates).
+    appConfig.numUniqueKeysToWrite = NUM_UNIQUE_KEYS;
+  }
+
+  private static final Map<String, Integer> rowsCount = new HashMap<>();
+  private static final Map<String, Integer> valuesPerRow = new HashMap<>();
+
+  private static final Map<Integer, Map<String, PreparedStatement>> preparedStatements = new HashMap<>();
+  private static final Map<Integer, Connection> connectionMap = new HashMap<>();
+
+  private static final Map<Integer, Integer> nextInt = new HashMap<>();
+  private static final Map<Integer, Integer> nextIntMax = new HashMap<>();
+  private static final Map<Integer, Long> nextLong = new HashMap<>();
+  private static final Map<Integer, Long> nextLongMax = new HashMap<>();
```

Comment:
These maps are extremely confusing. Are they using the same index, the thread id?
Perhaps we could replace them with an array where the values are structs (instances of a static inner class) containing prepared statement, connection, and various next int/long values as fields.